File: app/src/main/java/ua/com/sober/getwords/mvp/models/ms/ArrayOfGetTranslationsResponse.java
package ua.com.sober.getwords.mvp.models.ms;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

/**
 * Created by dmitry on 9/27/16.
 */

@Root(name = "ArrayOfGetTranslationsResponse")
public class ArrayOfGetTranslationsResponse {

    @ElementList(name = "GetTranslationsResponse", entry = "GetTranslationsResponse", inline = true)
    private List<GetTranslationsResponse> getTranslationsResponses;


    public List<GetTranslationsResponse> getGetTranslationsResponses() {
        return this.getTranslationsResponses;
    }

    public void setGetTranslationsResponses(List<GetTranslationsResponse> getTranslationsResponses) {
        this.getTranslationsResponses = getTranslationsResponses;
    }


    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this);
    }


    public static class GetTranslationsResponse {

        @Element(name = "From")
        private String from;

        @Element(name = "State", required = false)
        private Integer state;

        @Element(name = "Translations")
        private Translations translations;


        public String getFrom() {
            return this.from;
        }

        public void setFrom(String from) {
            this.from = from;
        }


        public Integer getState() {
            return this.state;
        }

        public void setState(Integer state) {
            this.state = state;
        }


        public Translations getTranslations() {
            return this.translations;
        }

        public void setTranslations(Translations translationses) {
            this.translations = translationses;
        }


        @Override
        public String toString() {
            return ToStringBuilder.reflectionToString(this);
        }

    }


    public static class Translations {

        @ElementList(name = "TranslationMatch", entry = "TranslationMatch", inline = true)
        private List<TranslationMatch> translationMatch;


        public List<TranslationMatch> getTranslationMatch() {
            return this.translationMatch;
        }

        public void setTranslationMatch(List<TranslationMatch> translationMatches) {
            this.translationMatch = translationMatches;
        }


        @Override
        public String toString() {
            return ToStringBuilder.reflectionToString(this);
        }

    }


    public static class TranslationMatch {

        @Element(name = "Error", required = false)
        private String error;

        @Element(name = "Count")
        private Integer count;

        @Element(name = "MatchDegree")
        private Integer matchDegree;

        @Element(name = "MatchedOriginalText", required = false)
        private String matchedOriginalText;

        @Element(name = "Rating")
        private Integer rating;

        @Element(name = "TranslatedText")
        private String translatedText;


        public String getError() {
            return error;
        }

        public void setError(String error) {
            this.error = error;
        }


        public Integer getCount() {
            return this.count;
        }

        public void setCount(Integer count) {
            this.count = count;
        }


        public Integer getMatchDegree() {
            return this.matchDegree;
        }

        public void setMatchDegree(Integer matchDegree) {
            this.matchDegree = matchDegree;
        }


        public String getMatchedOriginalText() {
            return this.matchedOriginalText;
        }

        public void setMatchedOriginalText(String matchedOriginalText) {
            this.matchedOriginalText = matchedOriginalText;
        }


        public Integer getRating() {
            return this.rating;
        }

        public void setRating(Integer rating) {
            this.rating = rating;
        }


        public String getTranslatedText() {
            return this.translatedText;
        }

        public void setTranslatedText(String translatedText) {
            this.translatedText = translatedText;
        }


        @Override
        public String toString() {
            return ToStringBuilder.reflectionToString(this);
        }

    }

}

File: app/src/main/java/ua/com/sober/getwords/mvp/models/ms/ArrayOfGetTranslationsResponse.kt
package ua.com.sober.getwords.mvp.models.ms

import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "ArrayOfGetTranslationsResponse")
class ArrayOfGetTranslationsResponse {
    @field:ElementList(name = "GetTranslationsResponse", entry = "GetTranslationsResponse", inline = true)
    var getTranslationsResponses: List<GetTranslationsResponse>? = null

    class GetTranslationsResponse {
        @field:Element(name = "From")
        var from: String? = null

        @field:Element(name = "State", required = false)
        var state: Int? = null

        @field:Element(name = "Translations")
        var translations: Translations? = null

        override fun toString(): String {
            return "GetTranslationsResponse(from=$from, state=$state, translations=$translations)"
        }
    }


    class Translations {
        @field:ElementList(name = "TranslationMatch", entry = "TranslationMatch", inline = true)
        var translationMatch: List<TranslationMatch>? = null

        override fun toString(): String {
            return "Translations(translationMatch=$translationMatch)"
        }
    }


    class TranslationMatch {
        @field:Element(name = "Error", required = false)
        var error: String? = null

        @field:Element(name = "Count")
        var count: Int? = null

        @field:Element(name = "MatchDegree")
        var matchDegree: Int? = null

        @field:Element(name = "MatchedOriginalText", required = false)
        var matchedOriginalText: String? = null

        @field:Element(name = "Rating")
        var rating: Int? = null

        @field:Element(name = "TranslatedText")
        var translatedText: String? = null

        override fun toString(): String {
            return "TranslationMatch(error=$error, count=$count, matchDegree=$matchDegree, matchedOriginalText=$matchedOriginalText, rating=$rating, translatedText=$translatedText)"
        }
    }

    override fun toString(): String {
        return "ArrayOfGetTranslationsResponse(getTranslationsResponses=$getTranslationsResponses)"
    }
}